Source/sound.cpp

```diff
@@ -261,82 +260,69 @@ TSnd *__fastcall sound_file_load(char *path)
 
 void __fastcall sound_CreateSoundBuffer(TSnd *sound_file)
 {
-	TSnd *v1; // esi
-	int v2; // eax
-	DSBUFFERDESC v3; // [esp+4h] [ebp-14h]
-
-	v1 = sound_file;
-	memset(&v3, 0, sizeof(DSBUFFERDESC));
-	v3.dwBufferBytes = v1->len;
-	v3.lpwfxFormat = &v1->fmt;
-	v3.dwSize = sizeof(DSBUFFERDESC);
-	v3.dwFlags = DSBCAPS_CTRLVOLUME|DSBCAPS_CTRLPAN|DSBCAPS_STATIC;
-	v2 = sglpDS->CreateSoundBuffer(&v3, &v1->DSB, NULL);
-	if ( v2 )
-		DSErrMsg(v2, 282, "C:\\Src\\Diablo\\Source\\SOUND.CPP");
+	DSBUFFERDESC DSB;
+	memset(&DSB, 0, sizeof(DSBUFFERDESC));
+	DSB.dwBufferBytes = sound_file->len;
+	DSB.lpwfxFormat = &sound_file->fmt;
+	DSB.dwSize = sizeof(DSBUFFERDESC);
+	DSB.dwFlags = DSBCAPS_CTRLVOLUME|DSBCAPS_CTRLPAN|DSBCAPS_STATIC;
+	int soundBuffer = sglpDS->CreateSoundBuffer(&DSB, &sound_file->DSB, NULL);
+
+	if ( soundBuffer )
+		DSErrMsg(soundBuffer, 282, "C:\\Src\\Diablo\\Source\\SOUND.CPP");
 }
 
 void __fastcall sound_file_cleanup(TSnd *sound_file)
 {
-	TSnd *v1; // esi
-	IDirectSoundBuffer *v2; // eax
-
-	v1 = sound_file;
-	if ( sound_file )
-	{
-		v2 = sound_file->DSB;
-		if ( v2 )
-		{
+	if ( sound_file ) {
+		if ( sound_file->DSB ) {
 			sound_file->DSB->Stop();
-			v1->DSB->Release();
-			v1->DSB = 0;
+			sound_file->DSB->Release();
+			sound_file->DSB = 0;
 		}
-		mem_free_dbg(v1);
+
+		mem_free_dbg(sound_file);
 	}
 }
 
 void __fastcall snd_init(HWND hWnd)
 {
 	sound_load_volume("Sound Volume", &sglSoundVolume);
 	gbSoundOn = sglSoundVolume > -1600;
+
 	sound_load_volume("Music Volume", &sglMusicVolume);
 	gbMusicOn = sglMusicVolume > -1600;
+
 	if ( sound_DirectSoundCreate(NULL, &sglpDS, NULL) )
 		sglpDS = 0;
+
 	if ( sglpDS && !sglpDS->SetCooperativeLevel(hWnd, DSSCL_EXCLUSIVE) )
 		sound_create_primary_buffer(0);
+
 	SVidInitialize(sglpDS);
 	SFileDdaInitialize(sglpDS);
+
 	gbSndInited = sglpDS != 0;
 }
 // 4A22D4: using guessed type char gbMusicOn;
 // 4A22D5: using guessed type char gbSoundOn;
 
 void __fastcall sound_load_volume(char *value_name, int *value)
 {
-	int *v2; // esi
-	//int v3; // eax
 	int v4; // ecx

```

Comment:
Could name this "vol" just to tidy up

Reply:
Actually I baled on that one, it's definatly not bin exact so shoudn't have been committed.